pytorch_lightning/core/step_result.py

```diff
@@ -0,0 +1,327 @@
+from typing import Optional, Dict, Union
+from torch import Tensor
+import torch
+from copy import copy
+
+
+class Result(Dict):
```

Comment:
docs also in the next PR?

Reply:
I feel like we should not add anything without docs

Reply:
this is not meant to be used right now. this is still in development. just breaking it up over 4 PRs because it's huge